src/broadcast.rs

```diff
@@ -70,61 +70,64 @@
 //! // Create a random set of keys for testing.
 //! let netinfos = NetworkInfo::generate_map(0..NUM_NODES);
 //!
-//! // Create initial nodes by instantiating a `Broadcast` for each:
-//! let mut nodes: BTreeMap<_, _> = netinfos.into_iter().map(|(i, netinfo)| {
-//!     let bc = Broadcast::new(Arc::new(netinfo), PROPOSER_ID)
-//!                  .expect("could not instantiate Broadcast");
-//!
-//!     (i, bc)
-//! }).collect();
+//! // Create initial nodes by instantiating a `Broadcast` for each.
+//! let mut nodes: BTreeMap<_, _> = netinfos
+//!     .into_iter()
+//!     .map(|(i, netinfo)| {
+//!         let bc =
+//!             Broadcast::new(Arc::new(netinfo), PROPOSER_ID).expect("Broadcast instantiation");
+//!         (i, bc)
+//!     })
+//!     .collect();
 //!
-//! // We are ready to start. First we generate a payload to broadcast:
+//! // We are ready to start. First we generate a payload to be broadcast.
 //! let mut payload: Vec<_> = vec![0; 128];
 //! rng.fill_bytes(&mut payload[..]);
 //!
-//! // Now we can start the algorithm, its input is the payload to be broadcast.
-//! let mut next_message = {
-//!        let proposer = nodes.get_mut(&PROPOSER_ID).unwrap();
-//!        proposer.input(payload.clone()).unwrap();
-//!
-//!        // attach the sender to the resulting message
-//!        proposer.next_message().map(|tm| (PROPOSER_ID, tm))
+//! // Now we can start the algorithm, its input is the payload.
+//! let initial_step = {
+//!     let proposer = nodes.get_mut(&PROPOSER_ID).unwrap();
+//!     proposer.input(payload.clone()).unwrap()
 //! };
 //!
-//! // We can sanity-check that a message is scheduled by the proposer:
-//! assert!(next_message.is_some());
+//! // Initialize the message queue for the simulated network. Annotate the resulting messages with
+//! // the sender ID.
+//! let mut messages: VecDeque<(_, TargetedMessage<_, _>)> = initial_step
+//!     .messages
+//!     .into_iter()
+//!     .map(|tm| (PROPOSER_ID, tm))
+//!     .collect();
+//!
+//! // We can check that a message is scheduled by the proposer. There should be one as long as the
```

Comment:
This sentence is a little confusing:  
There should be one as long as the number of nodes is greater than 1. 
I assume one is referring to a scheduled message, perhaps clarify.

Reply:
i.e. A scheduled message should exist if N > 1.

Reply:
This condition is implied by the other (newer) three. I'll remove it.